.github/check_license.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+usage() {
+  echo "$0 usage: [-r n] license file"
+  echo ""
+  echo "where:"
+  echo "-r n  max line number (1-based) where the first line of license header should start."
+  echo "      First line of the header must fall within range [1,n] (default: 1)"
+  echo ""
+  echo "SPECIAL TAGS: they can be used inside files to control behavior of this tool"
+  echo ""
+  echo "dlaf-no-check-license       skip any check and return 0"
+  echo "dlaf-multi-license-check    check license in the entire file (-r option is ignored)"
+
+  exit 0
+}
+
+MAX_LINE=1
+MULTILICENSE=false
+
+while getopts "hr:" arg; do
+  case $arg in
+    r)
+      MAX_LINE=$OPTARG
+      [ $MAX_LINE -lt 1 ] && echo "line number must be a value >= 1" && exit 1
+      ;;
+    h | *)
+      usage
+      ;;
+  esac
+done
+
+# note: check positional arguments
+[ $((OPTIND+1)) -gt $# ] && usage
+
+LICENSE_FILE=${@:$OPTIND:1}
+FILE_TO_CHECK=${@:$OPTIND+1:1}
+
+[ ! -f $LICENSE_FILE ]  && echo "$LICENSE_FILE not found."   && exit 1
+[ ! -f $FILE_TO_CHECK ] && echo "$FILE_TO_CHECK not found."  && exit 1
+
+
+# SPECIAL TAGs
+
+# note: skip the check
+grep -sq "dlaf-no-license-check" $FILE_TO_CHECK && exit 0
+
+# note: multi-license check
+if grep -sq "dlaf-multi-license-check" $FILE_TO_CHECK; then
+  MAX_LINE=`cat $FILE_TO_CHECK | wc -l | tr -d ' '`
+fi
```

Comment:
About special tags: it just looks for the tag, no matter where it is (comment, string, ...). Is it good enough?